samples/server/go-server/api/server/storage/memstore.go

```diff
@@ -162,6 +186,8 @@ func (m *memStore) ListNoteOccurrences(pID, nID, filters string) ([]*pb.Occurren
 	}
 	nName := name.FormatNote(pID, nID)
 	os := []*pb.Occurrence{}
+	m.RLock()
```

Comment:
There is a potential race here if another user deletes note between GetNote and m.RLock(). I wasn't sure how to properly do this without risking deadlocking. After reading up on RLock it seems to me that it should be ok to RLock before GetNote, whereas Lock would cause a deadlock.

Does RLocking before GetNote look ok to you @skelterjohn?

Reply:
Made the change in 4627a80. Let me know if you disagree.